sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystemProvider.java

```diff
@@ -1159,4 +1166,21 @@ private String extractAccountEndpoint(URI uri) {
 
         return endpoint;
     }
+
+    /**
+     * Sets the default configurations for this {@code FileSystemProvider}. The configurations will be applied if a call
+     * to {@link Paths#get(URI)} is made and there is not an open file system pointing to the specified endpoint. In
+     * this case, a corresponding file system will be opened. Otherwise, the already opened file system will be used.
+     * The endpoint must be specified in the uri.
+     *
+     * A file system will only be opened automatically if these defaults are set. If they are not set, attempting to
+     * retrieve a path in an unopened file system will continue to throw a {@link FileSystemNotFoundException}.
+     *
+     * @param config The configurations map. Please see the docs on {@link AzureFileSystemProvider for more information}
+     */
+    public static void setDefaultConfigurations(Map<String, Object> config) {
```

Comment:
is there any alternative design where this doesn't have to be a global ?

Reply:
I think the request was that it be global. What are your concerns around that? Perhaps I can find another way to address those and meet the customer ask.

Reply:
There's plenty of literature why global variables are dangerous. In this particular case the concern would be that users can modify this in flight while application is in the middle of processing workload which can potentially open a can of interesting race conditions for them.

On the other hand I understand the need for this convenience given that it's not the user interacting with types in this package but NIO magic doing it for them.

It seems the scenario here is to be able programmatically configure this at the startup of the application. I suggest we implement it in such a way that this configuration can be set only once during lifetime of user's program, so that they won't be tempted to modifying those settings in-flight.

Reply:
Makes sense. I can add an if statement to see if the values have already been set and throw if they have. I think that can satisfy the temptation to change things mid run, but if you are still more concerned about race conditions, I could try doing something with atomic values to prevent the chance of even very tight concurrent updates on startup. The latter strikes me as overkill personally

Reply:
"Double checked lock" might be good candidate for that.

Reply:
I'll look into that and put something up for your consideration.